lib/rules/no-inner-declarations.js

```diff
@@ -66,6 +76,22 @@ module.exports = {
 
     create(context) {
 
+        const sourceCode = context.sourceCode;
+        const programAst = sourceCode.ast;
+        const ecmaVersion = context.languageOptions.ecmaVersion || 6;
+        const options = context.options[1];
+        const legacy = options && options.legacy || false;
```

Comment:
We can make this a bit simpler.
```suggestion
        const legacy = context.options[1]?.legacy ?? false;
```